tower-overload/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+extern crate futures;
+extern crate tower_layer;
+extern crate tower_service;
+
+use futures::Poll;
+use tower_service::Service;
+
+pub mod error;
+mod future;
+mod layer;
+
+use self::error::Error;
+use self::future::ResponseFuture;
+pub use self::layer::OverloadLayer;
+
+#[derive(Debug)]
+pub struct Overload<S> {
+    inner: S,
+}
+
+// ===== impl Overload =====
+
+impl<S> Overload<S> {
+    pub fn new(inner: S) -> Self {
+        Overload { inner }
+    }
+}
+
+impl<S, Req> Service<Req> for Overload<S>
+where
+    S: Service<Req> + Clone,
+    S::Error: Into<Error>,
+{
+    type Response = S::Response;
+    type Error = Error;
+    type Future = ResponseFuture<S, Req>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        // We always report ready, and check inner `poll_ready` in the response
+        // future instead.
+        Ok(().into())
```

Comment:
I expected this implementation to call `poll_ready` on the inner service, and track if `Ready` was returned or not. It would always return `Ready` itself, but track the inner ret val for when `call` is invoked. If the inner service returned `Ready`, the request would be forwarded. If not, the request errors (overloaded).